src/sortStudents.ts

```diff
@@ -1,16 +1,87 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(grades: number[]): number {
+  return grades.reduce((sum, num) => sum + num, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const newStudents: Student[] = students
+    .map((person: Student) => ({ ...person }));
+
+  const isAscending = order === 'asc';
+
+  switch (sortBy) {
+    case SortType.Name: {
+      return newStudents.sort(isAscending
+        ? (student1: Student, student2: Student): number => (
+          student1.name.localeCompare(student2.name)
+        )
+        : (student1: Student, student2: Student): number => (
+          student2.name.localeCompare(student1.name)
+        ));
+    }
+
+    case SortType.Surname: {
+      return newStudents.sort(isAscending
+        ? (student1: Student, student2: Student): number => (
+          student1.surname.localeCompare(student2.surname)
+        )
+        : (student1: Student, student2: Student): number => (
+          student2.surname.localeCompare(student1.surname)
```

Comment:
```suggestion
        ? (student1, student2) => (
          student1.surname.localeCompare(student2.surname)
        )
        : (student1, student2) => (
          student2.surname.localeCompare(student1.surname)
```
You don't need to specify the type of callback function return value. (localeCompare always returns the number for strings).
Check code below